cmd/contour/servecontext.go

```diff
@@ -227,6 +236,14 @@ type LeaderElectionConfig struct {
 	Name          string        `yaml:"configmap-name,omitempty"`
 }
 
+// TimeoutConfig holds various configurable proxy timeout values.
+type TimeoutConfig struct {
+	// IdleTimeout defines how long the proxy should wait while there
+	// are no active requests before terminating an HTTP connection. Set
+	// to 0 to disable the timeout.
+	IdleTimeout time.Duration `yaml:"idle-timeout,omitempty"`
```

Comment:
@youngnick @jpeach @stevesloka before merging this, I'm considering renaming this field and related code to `ConnectionIdleTimeout` to more explicitly differentiate from `StreamIdleTimeout` which I'm adding next. What do you all prefer? `IdleTimeout` and `StreamIdleTimeout`, or `ConnectionIdleTimeout` and `StreamIdleTimeout`?

Reply:
Envoy has separate stream and connection timeouts, so we should name this for whichever Envoy timeout it refers to. [This](https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/timeouts) helps a lot.

Reply:
This one is the connection idle timeout, but envoy just calls it "idle_timeout".

Reply:
I'll leave this as-is so I can move ahead, but can always circle back if desired.

Reply:
I think `ConnectionIdleTimeout` & `StreamIdleTimeout` like you suggested is good. It helps to better clarify the context a bit.

Reply:
OK @stevesloka I updated the name, mind taking a quick look before I merge?

Reply:
> This one is the connection idle timeout, but envoy just calls it "idle_timeout".

Right. The doc I linked to explains that this is an alias to the stream timeout.